### ISSUE TITLE
[Feature Request] Import and export views

### ISSUE THREAD

**Issue by dellatino** [archimatetool/archi#270]
I'm fairly new to Archi and new here to the forum/github, so I apologize if this feature already exists and I haven't found it - feel free to point out whatever way you can think of :-)

In reference to the forum post:
http://forum.archimatetool.com/index.php?topic=304.0

I would love a way to migrate created diagrams (views) through different itterations of the Archi model I'm using. I have a few archimate files, (practically) the same model (read: almost no changes to the elements, changes mostly comprise of additions).
But today I'm always recreating the views from scratch or using a copy paste of the elements from the original model-view to the other model-view. Needless to say it's a hassle and it'll soon become impossible.

Possibly related to this issue: https://github.com/archimatetool/archi/issues/198. But since it's about a DB plugin there, I created this new issue.

Thank you for the efforts!

Regards
Z

**Comment by Phillipus** [comment]
I'm trying to boil this down...do you mean you'd like to have a View shared between different models? A model being a file-based entity?

**Comment by Phillipus** [comment]
Templates?

**Comment by dellatino** [comment]
Thank you for a quick response. I'll take a look if I missed something in regards to templates, but I don't see the relation yet.

For example purposes, let's say the model is the same, it's just a copy of the same file. But yesterday I created a view in one and today I need that view in the other one.

>Model
>> Strategy
> >Business
>>..
>>Other
>>Relations
>> **Views**
>>> A view called **Demo**, with a diagram

>Model_copy (from yesterday)
>> Strategy
>> Business
>>..
>>Other
>>Relations
>> **Views**
>>> I want the same **View** that's in my original Model under the name **Demo**

in Archi interface:
![model_copy](https://user-images.githubusercontent.com/31921376/30377507-39e634da-9890-11e7-8638-74dadebad065.png)

What I'm doing today:
create new view in Model_copy
name it Demo
go to first Model>Views>Demo
select all elements and copy
go to second Model>Views> newly created view Demo
paste
and wondering why I couldn't do this easier, since I have to do this more than once :)

Regards,
Z

**Comment by jbsarrodie** [comment]
@Phillipus Template does not work for that as they only allow to create a model using some pre-defined content, but here my understanding is to be able to move some content (elements, relations and views from one model to another).

So the answer is that this is not possible in Archi alone as this implies to extract all the content of a view and do a "merge" in another model (where a subset of the elements can already exist).

The good news (sort of) is that you can do it through the model repository and some Git-fu ;-)

Say for example you want to copy/move "view A" from "model 1" to "model 2". If both models are managed through the model repository (thus stored in a git repository) you can do the following:
* If not already done, commit all changes in "model 1"
* In "view A" select all elements and relationships and delete them **from the model**
* In "model 1" delete "view A"
* Commit "model 1" (this commit now contains the deletion of all elements of the view and related concepts)
* Select this last commit in the history tab and undo it using "Cancel up to this commit". This will in fact create a new commit containing the exact opposite actions (ie. create the view and all its concepts).
* Now use your favorite Git tool (e.g. SmartGit or SourceTree) and export this "undo" commit to a patch file.
* Use your favorite Git tool to apply this patch file on the repository containing "model 2"
* Open "model 2" in Archi through the model repository plugin, the view will be there.

Regards,

JB

**Comment by jbsarrodie** [comment]
@dellatino  A remark on your previous comment...

In addition to be boring to do it the way you do, there's another issue: if you want to copy 2 views that share some content, by creating 2 views in the target model and doing a copy/paste, there will no more shared concepts but several (different) concepts of the same type and with the same name.

Using the model repository and git would allow you to create a patch containing both views and shared content only once.

**Comment by dellatino** [comment]
A possible idea for a solution might be:
when you choose export to Open Exchange (or when you Import it again on the other file), an extra option 'export/import only archimate views'. Of course, this only gets you the same diagram if the references used on the diagram (Archimate views) are still intact.

But, that way you don't have to recreate the designed Archimate view.

Or am I missing something?

**Comment by Phillipus** [comment]
Templates - I meant to say, if one had re-usable templates of "patterns" or "snippets" of views and concepts. Not the current templates.

But I think this goes deeper to the questions of what is a "model" and should a "model" share components with other models. Perhaps there should just be a repository of "stuff" and this "stuff" is re-constituted and presented in different scopes (models)...

**Comment by dellatino** [comment]
@jbsarrodie spot on in regards to my question, thank you.
I'm already trying to understand how the Git-fu works, I need it internally / shouldn't go over a network..but it's a good start.

And I'll need to think about your last comment, I miss some knowledge to understand my second issue there :-)

Thx!

**Comment by herve91** [comment]
Hi,

Alternatively, you may use my database plugin (https://github.com/archi-contribs/database-plugin/tree/master/v2).

If it is for your unique usage,  the easiest way is to create a sqlite database.

You need to export your first model into the database, and then import your view In your others models.

You will be able to import the view in shared mode (all the updates done on this view will be seen in all the others models sharing this view) or in copy mode (the updates will remain local to your view)

Best regards
Hervé
